src/h3lib/lib/h3Index.c

```diff
@@ -340,17 +337,11 @@ int H3_EXPORT(compact)(const H3Index* h3Set, H3Index* compactedSet,
             }
         }
         // Set up for the next loop
-        for (int i = 0; i < numRemainingHexes; i++) {
-            hashSetArray[i] = 0;
-        }
-        free(remainingHexes);
+        memset(hashSetArray, 0, sizeof(hashSetArray));
         compactedSetOffset += uncompactableCount;
-        remainingHexes = compactableHexes;
+        memcpy(remainingHexes, compactableHexes, sizeof(compactableHexes));
         numRemainingHexes = compactableCount;
     }
-    // Final cleanup
-    free(remainingHexes);
-    free(hashSetArray);
     return 0;
```

Comment:
It looks like this line is no longer reached in tests, perhaps because of the change around line 276. Could you change to preserve 100% coverage?

Reply:
Sure. Will push fix for that now.